src/NuGet.Core/NuGet.Packaging/Signing/Signatures/Signature.cs

```diff
@@ -31,6 +31,11 @@ public class Signature
         /// </summary>
         public SignatureContent SignatureContent { get; }
 
+        /// <summary>
+        /// Signature timestamp.
+        /// </summary>
+        public Timestamp Timestamp { get; }
```

Comment:
I don't think the signature should have a reference to its Timestamp. In theory a signature could have 0..n timestamps. I think getting the timestamps should be a utility method instead of a property in the signature.

Reply:
Fair point on the 0..n on timestamps, how does the rest of the code even handle that today? From what I can see, I'm using the same mechanism to get a timestamp off the signature that the verification logic uses.

As for the 0 case, isn't that what `null` represents?

Reply:
I'm working to update this logic in the verify logic. You could see what I'm doing here https://github.com/NuGet/NuGet.Client/pull/1868/files#diff-ce902b6f1ff63e49285926ee427d5bf2R60 ... We could potentially merge this changes in one utility method that returns a list of timestamp objects or something related.

Reply:
The thing about having a timestamp as a property on the signature is that each timestamp (potentially unbounded list) can be valid or not valid depending in a bunch of offline and offline checks, and that in turn can be fatal or non fatal for different verification paths (e.g. verify command vs restore). So if we have a timestamp attribute in the signature we need to either have a possibly invalid timestamp or do the verification before needed. This is the reason I believe a utility method is a better approach.

Reply:
Yes, I do think we should make this property a `IReadOnlyList`. I think it's also a good idea to make it a lazy property, so that it is only loaded if it is needed, as the existence of timestamps is not required for use of the signature, it's just an extra bit of helpful information.